File: ksb-front/src/app/shared/models/locataire.ts
import { Garant } from "./garant";
import { Loyer } from "./loyer";
import { SituationProfessionnelle } from "./situation-professionnelle";
import { Utilisateur } from "./utilisateur";

export class Locataire extends Utilisateur {
    id!: number;
    nom!: string;
    prenom!: string;
    adresse!: string;
    telephone!: string;
    email!: string;
    profession!: string;
    numCompteBancaire !: String;
    situationProfessionnelle !: SituationProfessionnelle;
    garant!: Garant;
    loyer !: Loyer;
}

File: ksb-immo/src/main/java/sn/ksb/immo/ksbimmo/application/models/Locataire.java
package sn.ksb.immo.ksbimmo.application.models;

import lombok.*;

import javax.persistence.*;
import java.util.List;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
public class Locataire extends Utilisateur{

    private String nom;

    private String prenom;

    private String adresse;

    private String telephone;

    private String email;

    private String profession;

    @Embedded
    private SituationProfessionnelle situationProfessionnelle;

    @OneToOne(cascade = CascadeType.ALL)
    private Garant garant;

    private String numCompteBancaire;

    @ManyToMany
    @JoinColumn(table = "locataire_propriete")
    private List<Propriete> proprietes;

    @OneToOne(cascade = CascadeType.ALL)
    private Loyer loyer;

}
